### ISSUE TITLE
Event overlap not working in day/week/month view

### ISSUE THREAD

**Issue by timd2i** [fullcalendar/fullcalendar-scheduler#24]
Event overlap only seems to be working in the timelineview. Not in the other views. 

I used "eventOverlap:false".

http://jsbin.com/witimu/2/edit?js,output

**Comment by joshigauravv** [comment]
What is expected timeline to get a fix of this issue for paid customers?

**Comment by arshaw** [comment]
I apologize for the lag here. Thank you for the recreation. I will start the debugging process this week

**Comment by arshaw** [comment]
ahhh. this is happening because when scheduler is loaded, it enforces eventOverlap:false on a **per-resource** basis.

we should probably have `eventOverlap:false` always prevent range collisions, even across resources, and a new setting, probably called `resourceEventOverlap:false` that does it per-resource

**Comment by joshigauravv** [comment]
Hi Adam,
Though I did not debug the issue due to some reasons, ideally I think there
should be no need to check per resource/location. It should be unique per
instance of calendar.
May be it is getting reinitialised then to false which is default.

-Gaurav

**Comment by joshigauravv** [comment]
Hi Adam,
Greetings!
This is regarding the overlap issue on view other than day and 3 day. Just
wanted to check by when we can expect the fix for the overlap issue. We
have encountered one more issue where the grid lines disappears on FF
browser. Please find the attached screen shot.

Thanks,
Gaurav Joshi

**Comment by arshaw** [comment]
Hi Gaurav, the overlap fix will come in 1.1.0, which is about 2/3 weeks away given other project priorities. Is this okay?

Also, with your issue about grid lines disappearing, can you post that as a separate issue? The screenshots that you said you attached did not end up getting attached unfortunately. You might have to do is through the github interface (not email). Anyway, could you please repost this second issue regarding the lines?

**Comment by joshigauravv** [comment]
Thanks Adam for heads up.
After out consultation here, we would appreciate if it can be made
available by 6 Nov. 2015. We have a planned release in Nov. Also it has
been a while since it was reported.

Please accommodate to make it available by 6 Nov.

Thanks,
Gaurav Joshi

**Comment by arshaw** [comment]
Here is a quick fix. if you comment out these lines in the unminified source code, it should prevent event overlapping altogether when you have `eventOverlap` set to `false`

https://github.com/fullcalendar/fullcalendar-scheduler/blob/v1.0.2/dist/scheduler.js#L982-L995

**Comment by joshigauravv** [comment]
Thank you Adam for the solution.
But unfortunately the solution fixes the problem but creates a regression
on a day and 3 day view.

In this case no event can be dragged on any other location if a event is
present during that time period on some other location. So now overlapping
is w.r.t time but irrespective of locations.

Thanks,
Gaurav Joshi

**Comment by arshaw** [comment]
@sulian it did work i just tested it. here are the changes I made to get it to work, along with an HTML demo:

https://github.com/fullcalendar/fullcalendar-scheduler/pull/57/files

see the selectOverlap:false file. with `getPeerEvents` commented out, and with `selectable:true`, and with `selectOverlap:false`, selection is not allowed on ANY resource or day if there is an event during that time (on any resource).

can you double check?

**Comment by arshaw** [comment]
to get the raw scheduler.js file : https://raw.githubusercontent.com/fullcalendar/fullcalendar-scheduler/issue-24-hack/dist/scheduler.js

**Comment by arshaw** [comment]
@joshigauravv, i'm not sure what you want then. if you want to *disallow* events from overlapping in timelineThreeDay view (regardless of resource. only based on time), but you want to *allow* events to overlap on other views that don't explicitly display resources (like agendaDay and agendaThreeDay), that is **inconsistent** behavior across views. The user would theoretically be able to get around your `eventOverlap:false` in timeline by going to agendaDay, causing a collision, and then switching back. Am I getting this right?

**Comment by arshaw** [comment]
maybe i'm mixing up which view you do/don't want to disallow the resource-based overlapping, but the sentiment of my comment remains the same. you can't have one behavior in one view, and a different behavior in another.

**Comment by sulian** [comment]
@arshaw, the problem now is that you can't create any event in any resource if a range already exists : I cannot create an event for auditorium A because auditorium B have already an event...

**Comment by joshigauravv** [comment]
@Adam,
Yes, Sulian is correct. So once there is an event 3pm-4pm in auditorium A
then you can not create it on any other location/resource.

So the acceptable/required behavior is described below and it holds true
for all the view.
1. When a resource has an event from 3pm to 4pm then, it should not allow
to create the event on SAME resource in 3pm to 4pm slot. So overlap has to
be avoided. This works in a day and 3 day view with overlap = false. But on
DIFFERENT resource, user should be able to create an event even in 3pm-4pm.

2. Another observation is on week and month view since resource is not
visible, when an event is dragged, resource remains same. This is correct.
So ideally it should not create a problem but then it should check if the
overlap is happening by checking the resource id.

Let me know if it helps or require another exhaustive example/case.

-Gaurav

**Comment by arshaw** [comment]
I now understand! I'm really sorry for not understanding initially, my mind was stuck on another theory. i can now create your problem. i will get this fixed in the next release.

**Comment by sulian** [comment]
:+1:

**Comment by joshigauravv** [comment]
Hi Adam,
Greetings for the day!
With reference to our communication on Oct. 25. You had mentioned 2/3 weeks
timeframe of a formal release. I thought I will check with you if it can be
made available in coming week.
We have one of the major release planned and hoping to get this issue out.

Can you please update/inform about the plan for the issue/ticket?


Thanks,
Gaurav Joshi
+1 415 604 6280
[image: https://www.linkedin.com/in/joshigauravv]
<https://www.linkedin.com/in/joshigauravv>

**Comment by stillfire** [comment]
I found an issue with external events and the scheduler, adding an overlap:false does not work the same as it does in the full calendar - is this issue related or should I make a new one?

     $('#external-events .fc-event').each(function() {
				$(this).data('event', {
					title: $.trim($(this).text()), 
					stick: true, 
					overlap: false // HAS NO EFFECT - WORKS IN FULLCALENDAR BUT NOT SCHEDULER
				});
				$(this).draggable({
					zIndex: 999,
					revert: true,  
					revertDuration:
				});
			});

**Comment by arshaw** [comment]
Hi @joshigauravv I'm hoping to get this into `1.1-beta2`, to be release Monday

**Comment by arshaw** [comment]
this is fixed in a beta release. i've modified the original JSbin to show you that the problem is fixed:
http://jsbin.com/kesewoqupo/3/edit?html,output

it requires the latest beta releases of both scheduler and fullcalendar core.

to be released officially soon.

**Comment by arshaw** [comment]
https://github.com/fullcalendar/fullcalendar/releases/tag/v2.5.0-beta2
https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v1.1.0-beta2

**Comment by arshaw** [comment]
@stillfire to continue the conversation from the closed issue #72, are you sure? Have you tried this demo? http://jsbin.com/kesewoqupo/3/edit?html,output

It'd be very helpful if you could create a JSBin recreation of your problem. I was hoping I had completely solved it.

**Comment by arshaw** [comment]
@stillfire my mistake, let's resume the conversaion on #72. I didn't realize your report was dealing with *external* event dragging.

**Comment by arshaw** [comment]
fixed and release in [v1.1.0](https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v1.1.0)

@stillfire we can resume convo on #72 about your specific issue

**Comment by joshigauravv** [comment]
Hi Adam,
What I understand from the fix delivery is I need to chenage just the
scheduler code right? I need not change the full calendar code.
Can you please confirm?

Also when I do this I am getting error. Please suggest. Is there any change
in the property "resources"?

Thanks,
Gaurav Joshi
+1 415 604 6280
[image: https://www.linkedin.com/in/joshigauravv]
<https://www.linkedin.com/in/joshigauravv>

**Comment by arshaw** [comment]
@joshigauravv you will need to upgrade to fullcalendar core 2.5.0

**Comment by joshigauravv** [comment]
Hi Adam,
Thanks for the reply. As per your suggestion I tried to replace in
following fashion.
1. I replaced fullcalender.js
2. I replaced Scheduler.js

I did not update other files mostly stylesheets and momentjs. However I do
see the overlap problem is resolved while external drag is screwed up. It
did not pick up the start and end date because of which I see hair like
thin session when dragged from outside/externally. I believe not copying
the css should not impact this since momentJS and styles will be same.

Pls guide.

Thanks,
Gaurav Joshi
+1 415 604 6280
[image: https://www.linkedin.com/in/joshigauravv]
<https://www.linkedin.com/in/joshigauravv>
